### PR TITLE
Support multiple legend entries for same glyph renderer

### PR DESCRIPTION
**Pull request by birdsarah**
Fixes: #3856

Follows on from PR #5088

### CODE REVIEW

**Comment by birdsarah** [comment]
screen shot 2016-09-03 at 4 28 45 am

File: bokeh/plotting/helpers.py [hunk 1]
```diff
@@ -98,6 +99,27 @@ def _pop_colors_and_alpha(glyphclass, kwargs, prefix="", default_alpha=1.0):
...
     return result
 
+
+def _process_legend_kwargs(kwargs):
+    if kwargs.get('legend') and kwargs.get('label'):
+        raise RuntimeError("Cannot set both legend and label at the same time.")
+    legend = kwargs.pop('legend', None)
+    source = kwargs.get('source')
+    if legend:
+        if isinstance(legend, str):
```
Comment (mattpap) [inline bokeh/plotting/helpers.py]: `six.string_types`

Comment (birdsarah) [inline bokeh/plotting/helpers.py]: Thanks @mattpap, fixed.

**Comment by birdsarah** [comment]
(I'll deal with the conflict here once #5088 is merged)

**Comment by bryevdv** [comment]
@birdsarah the merge seemed straightforward, but please double check (I ran the User's Guide example and it worked great). Otherwise, LGTM to merge when green

**Pull request merged**